src/repositories/channelRepository.js

```diff
@@ -0,0 +1,90 @@
+const Message = require('../models/Message');
+const channels = [];
+
+const createChannel = (channelKey) => {
+    channels[channelKey] = {
+        users: [],
+        messages: [],
+    };
+};
+
+exports.channelList = () => (
+    Promise.resolve(Object.keys(channels))
+);
+
+exports.existChannel = (channelKey) => (
+    new Promise((resolve, _) => {
+        exports.channelList().then((channelList) => {
+            resolve(
+                !!channelList.find(c => c === channelKey)
+            );
+        });
+    })
+);
+
+exports.createChannel = async (channelKey) => {
+    const channelExist = await exports.existChannel(channelKey);
+    return new Promise((resolve, reject) => {
+        if (!channelExist) {
+            createChannel(channelKey);
+            resolve();
+        }
+        reject(`Channel ${channelKey} already exist.`);
+    });
+};
+
+exports.existUser = async (channelKey, userId) => {
+    const existChannel = await exports.existChannel(channelKey);
+    return new Promise((resolve, reject) => {
+        if (!existChannel) {
+            reject(`${channelKey} does not exist.`);
+        };
+        resolve(
+            !!channels[channelKey].users.find(u => u === userId)
```

Comment:
Are we covering here the cae where channelKey is an empty string?

Reply:
I've added new tests for this.